lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -31,7 +31,12 @@ def onNewMessage(ctx, message)
       hash = message.getData
       ip_address = ip_address(ctx)
 
-      hash['@metadata']['ip_address'] = ip_address unless ip_address.nil? || hash['@metadata'].nil?
+      unless ip_address.nil? || hash['@metadata'].nil?
+        tmp_evt = org.logstash.Event.new
+        tmp_evt.setField(@input.field_hostip, ip_address)
+        # Leverage Logstash's event to convert nested event fields into nested hashes
+        nested_fields_to_maps(hash, tmp_evt)
+      end
       target_field = extract_target_field(hash)
```

Comment:
this line has an effect of setting more `@metadata`, 
since we now use a common prefix for the other meta-data `[input][beats][...]`
we should consider also changing https://github.com/logstash-plugins/logstash-input-beats/blob/master/lib/logstash/inputs/beats/message_listener.rb#L143-L152 to use a prefix in ECS mode.